### DISCUSSION TITLE
How to add relation on microservices?

### DISCUSSION THREAD

**Discussion by binotby** [prisma/prisma#17751] [Q&A]
Schema on User Management service:
```
model User {
  id                 String    @id @default(uuid())
  username           String    @unique
  email              String    @unique
  password           String
}
```

Schema on File management service:
```
model File {
  id               String   @id @default(uuid())
  user_id          String
  user             User     @relation(fields: [user_id], references: [id])
  file_name        String
  file_description String
  file_url         String
  file_hash        String
}
```

My user management service and file management service is different repository using Nest Js, Thank You!

**Comment by nurul3101** [comment]
Hello @binotby 👋 
Do you mean you have multiple prisma schema files, and you want to have a relation that is defined among them?

On a sidenote, we have examples of using NestJS and Prisma [here](https://github.com/prisma/prisma-examples/tree/latest/typescript) and also a detailed [guide](https://www.prisma.io/blog/nestjs-prisma-rest-api-7D056s1BmOL0) that could be useful to you.

**Reply by binotby** [reply]
Hi @nurul3101 yes i have multiple schema files and i want to have a relation, but in file model the user is undifined because the user model on another repository

**Reply by nurul3101** [reply]
Which tool are you using to split the schema files? As prisma doesn't support splitting schema natively
- https://github.com/prisma/prisma/issues/2377

**Reply by binotby** [reply]
can i split prisma schema on another repository? @nurul3101

**Reply by nurul3101** [reply]
You would need to use community-built tools like Prisma Import to split schema files: https://github.com/ajmnz/prisma-import

**Comment by binotby** [comment]
i think we need this feature https://docs.djangoproject.com/en/4.1/ref/models/options/#django.db.models.Options.managed

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
